src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1299,8 +1300,9 @@ protected void GenerateFindFirstChar()
                             Ret();
                             break;
 
-                        case 2:
-                        case 3:
+                        default:
+                            Debug.Assert(setCharsCount == 2 || setCharsCount == 3, $"Unexpected setCharsCount: {setCharsCount}");
```

Comment:
aside, where is the place it is limiting to 3 chars here? I can't find it.

Reply:
The size of the stackalloc above.

Reply:
Ah I see -- and that is passed to GetSetChars, which will return 0 if it can't fit them in the array, which will cause the condition to be false. It might be worth a comment on the stackalloc that the "3" there is the magic number determining whether it uses the indexof path.

Reply:
Ok, I'll add a comment in a subsequent PR to avoid restarting CI for it.